src/uct/sm/mm/mm_iface.c

```diff
@@ -403,8 +403,12 @@ static void uct_mm_iface_recv_messages(uct_mm_iface_t *iface)
         ret = recvfrom(iface->signal_fd, &sig, sizeof(sig), 0, NULL, 0);
         if (ret == sizeof(sig)) {
             ucs_debug("mm_iface %p: got connect message", iface);
-            ucs_callbackq_add_safe(&iface->super.worker->progress_q,
-                                   uct_mm_iface_progress, iface);
+            if (iface->super.prog.refcount++ == 0) {
```

Comment:
1. ucs_callbackq_add_safe is currently the same as ucs_callbackq_add, but the use of callback should not take advantage of that knowldge, since according to API _safe version should be used if calling from another thread
1. ASYNC_BLOCK is not needed from the async context itself